server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java

```diff
@@ -1447,11 +1470,12 @@ public synchronized void close() {
 
       closed = true;
 
-      // wait while internal jobs are running or external compactions are committing, but do not
-      // wait on external compactions that are running
+      // wait while internal jobs are running, external compactions are committing or the status of
+      // chops is active, but do not wait on external compactions that are running
       while (runningJobs.stream()
           .anyMatch(job -> !((CompactionExecutorIdImpl) job.getExecutor()).isExternalId())
-          || !externalCompactionsCommitting.isEmpty()) {
+          || !externalCompactionsCommitting.isEmpty()
+          || fileMgr.chopStatus != ChopSelectionStatus.NOT_ACTIVE) {
```

Comment:
Need to make this check more narrow.  After closed is set to true above, its not possible to enter the marking state. So we only want to wait on things that were in the marking state before closed was set to true.
  
```suggestion
          || fileMgr.chopStatus == ChopSelectionStatus.MARKING) {
```

Reply:
Oops, I had thought I included this change earlier. Just included in 2e3d97e